plugins/regions.js

```diff
@@ -147,34 +152,52 @@ window.plugin.regions.getSearchResult = function(match) {
   var regionI = id1-1;
   var regionJ = codeWordId;
 
-  var result = {}, level;
+  // the final 2 bits of I and J are tricky - easiest way is to try all 16 cells given the 4 bits of I/J we have so far...
+
+  if(id2 === undefined) { // level 4 cell
+    // as in the name-construction above, for odd numbered faces, the I and J need swapping
+    var cell = (faceId == 1 || faceId == 3 || faceId == 5) ?
+      S2.S2Cell.FromFaceIJ(faceId, [regionJ,regionI], 4) :
+      S2.S2Cell.FromFaceIJ(faceId, [regionI,regionJ], 4) ;
+    result.title = window.plugin.regions.FACE_NAMES[faceId] + zeroPad(id1,2) + '-' + window.plugin.regions.CODE_WORDS[codeWordId];
+
+    var corners = cell.getCornerLatLngs();
+    result.layer = L.geodesicPolygon(corners, { fill: false, color: 'red', clickable: false });
 
-  if(id2 === undefined) {
-    result.description = 'Regional score cells (cluster of 16 cells)';
+    result.bounds = L.latLngBounds(corners);
     result.icon = 'data:image/svg+xml;base64,'+btoa('@include_string:images/icon-cell.svg@'.replace(/orange/, 'gold'));
-    level = 4;
-  } else {
-    result.description = 'Regional score cell';
-    result.icon = 'data:image/svg+xml;base64,'+btoa('@include_string:images/icon-cell.svg@');
-    level = 6;
-
-    var xy = window.plugin.regions.d2xy(4, id2);
```

Comment:
this line is indeed wrong, the position of the id2 cell depends on the region (cluster) position.
maybe a clever use of `rot` with the correct arguments can solve this (to avoid a loop)

the loop is good enough tho